tests/integration/test_integration.py

```diff
@@ -535,6 +801,608 @@ def test_uniques_subset_within(engine, unique_table1, data):
     assert operation(test_result.outcome), test_result.failure_message
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            negation,
+            ["col_int", "col_varchar"],
+            [(0, "hi0"), (1, "hi0")],
+            0,
+            None,
+            False,
+            util_output_postprocessing_sorter,
+            slice(5),
+            None,
+            None,
+            "column(s) 'col_int', 'col_varchar' has a fraction of 0.9516129032258065 > 0 values (59 / 62) "
+            "not being an element of '[(0, 'hi0'), (1, 'hi0')]'. "
+            "It has e.g. (slice(None, 5, None)) excess elements "
+            "'[(2, 'hi1'), (3, 'hi2'), (5, 'hi3'), (6, 'hi4'), (8, 'hi5')]' with counts [2, 2, 2, 2, 2].",
+        ),
+        (
+            negation,
+            ["col_int", "col_varchar"],
+            [(0, "hi0"), (1, "hi0")],
+            0,
+            None,
+            True,
+            util_output_postprocessing_sorter,
+            slice(None),
+            None,
+            None,
+            "column(s) 'col_int', 'col_varchar' has a fraction of 0.9523809523809523 > 0 DISTINCT values "
+            "(40 / 42) not being an element of '[(0, 'hi0'), (1, 'hi0')]'. "
+            "It has e.g. (slice(None, None, None)) excess elements "
+            "'[(2, 'hi1'), (3, 'hi2'), (5, 'hi3'), (6, 'hi4'), (8, 'hi5'), (9, 'hi6'), (11, 'hi7'), (12, "
+            "'hi8'), (14, 'hi9'), (15, 'hi10'), (17, 'hi11'), (18, 'hi12'), (20, 'hi13'), (21, 'hi14'), (23, "
+            "'hi15'), (24, 'hi16'), (26, 'hi17'), (27, 'hi18'), (29, 'hi19'), (None, None), (None, 'hi'), "
+            "(1, 'hi1'), (4, 'hi2'), (4, 'hi3'), (7, 'hi4'), (7, 'hi5'), (10, 'hi6'), (10, 'hi7'), (13, "
+            "'hi8'), (13, 'hi9'), (16, 'hi10'), (16, 'hi11'), (19, 'hi12'), (19, 'hi13'), (22, 'hi14'), (22, "
+            "'hi15'), (25, 'hi16'), (25, 'hi17'), (28, 'hi18'), (28, 'hi19')]' with counts [2, 2, 2, 2, 2, "
+            "2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, "
+            "1, 1, 1].",
+        ),
+        (
+            negation,
+            ["col_int"],
+            [
```

Comment:
Maybe we instruct our linter/formatter not to do one element per line in this list? Is really unnecessarily bloating the line count. Same for the other parameterizations.

Alternatively, perhaps you can do something like `list(range(30))` instead.